File: designacoes/src/main/java/com/example/domains/Publicador.java
package com.example.domains;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
public class Publicador {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany
    private List<Relatorio> relatorios;

    @Column(nullable = false, length = 150)
    private String nome;

    @Column
    private Enum sexo;

    @Column
    private Boolean flBatismo;

    @Column
    private Enum perfil;

    @Column
    private Enum publicadorTipo;

    @Column
    private String observacao;
}

File: designacoes-front/src/app/shared/domain/publicador.ts
import { Relatorio } from "./relatorio";

export class Publicador {
    id : number = 0;
    nome: string = "";  
    sexo: string = "";  
    flBatismo: boolean = true;  
    perfil: string = "";  
    publicadorTipo: string = "";  
    observacao: string = ""; 
    relatorioDTO: Relatorio = new Relatorio('',0,0,false,0,0,0,0,0,''); 
 
    constructor() {}
 } 